crates/autopilot/src/infra/blockchain/circuit_breaker.rs

```diff
@@ -0,0 +1,131 @@
+use {
+    crate::{
+        domain,
+        infra::blockchain::{
+            contracts::{deployment_address, Addresses},
+            Rpc,
+        },
+    },
+    ethcontract::transaction::TransactionResult,
+    primitive_types::H160,
+    std::{sync::Arc, time::Duration},
+    url::Url,
+};
+
+// Time between Ethereum blocks: 12 seconds
+// In order to fulfill Nyquist theory about sampling rate, we need to at least
+// sample twice the give frequency, in other words, half the period
+const SETTLEMENT_CHECK_SAMPLE_RATE: Duration = Duration::from_secs(6);
```

Comment:
We don't need to poll for this. You can subscribe to the `Settlement` event and automatically get notified when a new auction happened.

Reply:
Ahh, I didn't know that, very cool. Thank you!